master/internal/trial.go

```diff
@@ -452,58 +475,67 @@ func (t *trial) allocationExited(ctx *actor.Context, exit *task.AllocationExited
 			return err
 		}
 		if t.restarts > t.config.MaxRestarts() {
-			return t.transition(ctx, model.ErrorState)
+			return t.transition(ctx, model.StateWithReason{
+				State:               model.ErrorState,
+				InformationalReason: "trial exceeded max restarts",
+			})
 		}
 	case exit.UserRequestedStop:
 		ctx.Tell(ctx.Self().Parent(), trialReportEarlyExit{
 			requestID: t.searcher.Create.RequestID,
 			reason:    model.UserCanceled,
 		})
-		return t.transition(ctx, model.CompletedState)
+		return t.transition(ctx, model.StateWithReason{
+			State:               model.CompletedState,
+			InformationalReason: "trial completed due to a user requested stop",
+		})
 	case t.userInitiatedExit != nil:
 		ctx.Tell(ctx.Self().Parent(), trialReportEarlyExit{
 			requestID: t.searcher.Create.RequestID,
 			reason:    *t.userInitiatedExit,
 		})
-		return t.transition(ctx, model.CompletedState)
+		return t.transition(ctx, model.StateWithReason{
+			State:               model.CompletedState,
+			InformationalReason: "trial completed due to a user initiated exit",
```

Comment:
"trial exited early with reason: %v", *t.userInitiatedExit.reason (or wherever that field is).

the reason is pretty interesting, so we should def include it.